cmake/EthCompilerSettings.cmake

```diff
@@ -18,103 +18,110 @@ include(EthCheckCXXCompilerFlag)
-	set(CMAKE_CXX_FLAGS_DEBUG          "-Og -g -DETH_DEBUG")
-	set(CMAKE_CXX_FLAGS_MINSIZEREL     "-Os -DNDEBUG")
-	set(CMAKE_CXX_FLAGS_RELEASE        "-O3 -DNDEBUG")
-	set(CMAKE_CXX_FLAGS_RELWITHDEBINFO "-O2 -g")
-
-	option(USE_LD_GOLD "Use GNU gold linker" ON)
-	if (USE_LD_GOLD)
-		execute_process(COMMAND ${CMAKE_C_COMPILER} -fuse-ld=gold -Wl,--version ERROR_QUIET OUTPUT_VARIABLE LD_VERSION)
-		if ("${LD_VERSION}" MATCHES "GNU gold")
-			set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -fuse-ld=gold")
-			set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -fuse-ld=gold")
-		endif ()
-	endif ()
-
-	# Check GCC compiler version.
-	if ("${CMAKE_CXX_COMPILER_ID}" MATCHES "GNU")
-		if("${CMAKE_CXX_COMPILER_VERSION}" VERSION_LESS 4.8)
-			message(FATAL_ERROR "This compiler ${CMAKE_CXX_COMPILER_ID} ${CMAKE_CXX_COMPILER_VERSION} is not supported. GCC 4.8 or newer is required.")
-		endif()
-
-	# Stop if buggy clang compiler detected.
-	elseif("${CMAKE_CXX_COMPILER_ID}" MATCHES AppleClang)
-		if("${CMAKE_CXX_COMPILER_VERSION}" VERSION_LESS 8.4)
-			message(FATAL_ERROR "This compiler ${CMAKE_CXX_COMPILER_ID} ${CMAKE_CXX_COMPILER_VERSION} is not able to compile required libff. Install clang 4+ from Homebrew or XCode 9.")
-		endif()
-	endif()
+    # Enables all the warnings about constructions that some users consider questionable,
+    # and that are easy to avoid.  Also enable some extra warning flags that are not
+    # enabled by -Wall.   Finally, treat at warnings-as-errors, which forces developers
+    # to fix warnings as they arise, so they don't accumulate "to be fixed later".
+    add_compile_options(-Wall)
+    add_compile_options(-Wextra)
+    add_compile_options(-Werror)
+
+    # Disable warnings about unknown pragmas (which is enabled by -Wall).
+    add_compile_options(-Wno-unknown-pragmas)
+
+    # Configuration-specific compiler settings.
+    set(CMAKE_CXX_FLAGS_DEBUG          "-Og -g -DETH_DEBUG")
+    set(CMAKE_CXX_FLAGS_MINSIZEREL     "-Os -DNDEBUG")
+    set(CMAKE_CXX_FLAGS_RELEASE        "-O3 -DNDEBUG")
+    set(CMAKE_CXX_FLAGS_RELWITHDEBINFO "-O2 -g")
+
+    option(USE_LD_GOLD "Use GNU gold linker" ON)
+    if (USE_LD_GOLD)
+        execute_process(COMMAND ${CMAKE_C_COMPILER} -fuse-ld=gold -Wl,--version ERROR_QUIET OUTPUT_VARIABLE LD_VERSION)
+        if ("${LD_VERSION}" MATCHES "GNU gold")
+            set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -fuse-ld=gold")
+            set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -fuse-ld=gold")
+        endif ()
+    endif ()
+
+    # Hide all symbols by default.
+    add_compile_options(-fvisibility=hidden)
+    if(CMAKE_SYSTEM_NAME STREQUAL "Linux")
+        # Do not export symbols from dependencies.
+        set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,--exclude-libs,ALL")
+    endif()
+
+    # Check GCC compiler version.
+    if ("${CMAKE_CXX_COMPILER_ID}" MATCHES "GNU")
+        if("${CMAKE_CXX_COMPILER_VERSION}" VERSION_LESS 4.8)
```

Comment:
We don't support 4.8 anymore, maybe we should change it in this PR

Reply:
Will do. Anything else?